### ISSUE TITLE
Brakeman audit on current codebase

### ISSUE THREAD

**Issue by ang-st** [openware/peatio#336]
Hello, 

i gave a shot at https://github.com/presidentbeef/brakeman to see if it would find some potential security issues. 
And there are some match : 

```
ubuntu@ip-xxx:~/rubykube/peatio$ brakeman 
Loading scanner...
[...]
== Brakeman Report ==

Application Path: /home/ubuntu/rubykube/peatio
Rails Version: 4.2.10
Brakeman Version: 4.1.1
Scan Date: 2018-01-22 16:23:48 +0000
Duration: 6.205497419 seconds
Checks Run: BasicAuth, BasicAuthTimingAttack, ContentTag, CreateWith, CrossSiteScripting, DefaultRoutes, Deserialize, DetailedExceptions, DigestDoS, DynamicFinders, EscapeFunction, Evaluation, Execute, FileAccess, FileDisclosure, FilterSkipping, ForgerySetting, HeaderDoS, I18nXSS, JRubyXML, JSONEncoding, JSONParsing, LinkTo, LinkToHref, MailTo, MassAssignment, MimeTypeDoS, ModelAttrAccessible, ModelAttributes, ModelSerialize, NestedAttributes, NestedAttributesBypass, NumberToCurrency, PermitAttributes, QuoteTableName, Redirect, RegexDoS, Render, RenderDoS, RenderInline, ResponseSplitting, RouteDoS, SQL, SQLCVEs, SSLVerify, SafeBufferManipulation, SanitizeMethods, SelectTag, SelectVulnerability, Send, SendFile, SessionManipulation, SessionSettings, SimpleFormat, SingleQuotes, SkipBeforeFilter, StripTags, SymbolDoSCVE, TranslateBug, UnsafeReflection, ValidationRegex, WithoutProtection, XMLDoS, YAMLParsing

== Overview ==

Controllers: 51
Models: 67
Templates: 133
Errors: 0
Security Warnings: 7

== Warning Types ==

Cross-Site Request Forgery: 1
Cross-Site Scripting: 1
Mass Assignment: 1
Redirect: 1
Remote Code Execution: 1
SQL Injection: 2

== Warnings ==

Confidence: High
Category: Cross-Site Request Forgery
Check: ForgerySetting
Message: 'protect_from_forgery' should be called in Test::ModuleController
File: app/controllers/test/module_controller.rb
Line: 2

Confidence: High
Category: Mass Assignment
Check: PermitAttributes
Message: Potentially dangerous key allowed for mass assignment
Code: params.require(:deposit).permit(:fund_source, :amount, :currency, :account_id, :member_id)
File: app/controllers/concerns/deposits/ctrl_bankable.rb
Line: 38

Confidence: High
Category: Redirect
Check: Redirect
Message: Possible unprotected redirect
Code: redirect_to(cookies[:redirect_to])
File: app/controllers/application_controller.rb
Line: 21

Confidence: Medium
Category: Remote Code Execution
Check: UnsafeReflection
Message: Unsafe reflection method constantize called with model attribute
Code: "CoinRPC::#{Currency.find_by_code(currency.to_s).code.upcase}".constantize
File: lib/coin_rpc.rb
Line: 8

Confidence: Medium
Category: SQL Injection
Check: SQL
Message: Possible SQL injection
Code: self.class.connection.execute("update accounts set balance = balance + #{delta_b}, locked = locked + #{delta_l} where id = #{id}")
File: app/models/account.rb
Line: 158

Confidence: Medium
Category: SQL Injection
Check: SQL
Message: Possible SQL injection
Code: connection.insert("INSERT INTO account_versions (#{attrs.symbolize_keys.keys.join(",")}) #{Account.unscoped.select(ActiveRecord::Base.sanitize_sql_array([(["?"] * attrs.symbolize_keys.size).join(","), *attrs.symbolize_keys.values])).where(:id => attrs.symbolize_keys[:account_id], :balance => balance, :locked => locked).to_sql}")
File: app/models/account_version.rb
Line: 54

Confidence: Weak
Category: Cross-Site Scripting
Check: CrossSiteScripting
Message: Unescaped model attribute
Code: t(".description", Currency.assets("usd")["accounts"].first.symbolize_keys)
File: app/views/private/assets/_usd_assets.html.erb
Line: 24

```
I didn't dive further to test how exploitable are each issues, so those might be false positives

Best

**Comment by yivo** [comment]
Hi @ang-st 

Thank you for your submit. We will validate all cases.

**Comment by yivo** [comment]
@vpetrusenko Please, work in devel branch.

**Comment by yivo** [comment]
Closed via #350.
